woodwork/type_sys/utils.py

```diff
@@ -22,6 +23,17 @@ def col_is_datetime(col, datetime_format=None):
     if len(col) == 0:
         return False
 
+    try:
+        if pd.api.types.is_numeric_dtype(pd.Series(col.values.tolist())):
```

Comment:
This will be fine computationally? Can't we just do:
```suggestion
        if pd.api.types.is_numeric_dtype(col):
```

Reply:
Yep, this is in response to an object edge-case:
<img width="255" alt="image" src="https://user-images.githubusercontent.com/22552445/166559848-cfb5ec58-b912-4a37-8ab7-a8fa2a468c6d.png">

We still want to infer this as numeric, and it turns out that the only way we can remove this dtype casting is by accessing the list of values via `tolist()`
<img width="409" alt="image" src="https://user-images.githubusercontent.com/22552445/166560046-2c075801-8304-4fa6-9bd8-803757d9bed6.png">

Reply:
alright